src/trio/_core/_run.py

```diff
@@ -2793,6 +2800,8 @@ def unrolled_run(
             ),
             stacklevel=1,
         )
+    except RuntimeWarning:
+        raise
     except TrioInternalError:
         raise
```

Comment:
This could be too broad (maybe warnings should cause internal errors and prompt people to open issues) or too narrow (if passing through runtimewarnings, why not all warnings?). Could make the `RuntimeWarning` a custom warning class instead

Reply:
Hmm, I'm torn on this - in large part because `TrioInternalError` prompts users to open an issue to report it, and running pytest with `-Werror` isn't that obscure

Reply:
Oh, I see!  Yeah, keeping it with a custom warning type seems good then.

Reply:
To be honest custom warning type is unnecessary if above `lot.break_lot(task)` we put a `lot.broken_by` check. I'm not sure that indicative of quality of implementation though.

Reply:
Requiring that check (and erroring on attempts to break an already-broken lot) seems pretty sensible to me, it's not like `ParkingLot` is an interface that gets called _directly_ very often, and so I don't mind adding some ceremony if that helps catch tricky concurrency bugs.

Reply:
okay so with the check we no longer raise any warnings on task exit if multiple tasks try to break a task, which means we only ever give a warning if manually breaking a broken lot. That doesn't seem particularly logical to me, and we should probably do one of:
1. make breaking a broken lot a no-op
2. raise an error
3. ...go with `broken_by` being a list (or tuple).

idc much which one though. The warning was added after this convo: https://github.com/python-trio/trio/pull/3081/files/7a1ce5b755b5775153a3525bea793ab7da05fb44#r1764544347 where I argued against the list, but if we no longer want to raise a warning from the internals then maybe the list is the way to go? (that also allows for duplicates in the list).

Reply:
I agree with that ordering (though I might slightly prefer 2 over 1? doesn't matter.)